src/server/renderer.js

```diff
@@ -44,14 +44,14 @@ export default (projectConfig, projectToolsConfig) => {
           console.error('ROUTER ERROR:', pretty.render(error));
           res.status(500);
         } else if (renderProps) {
-          loadPropsOnServer(renderProps, (err, asyncProps) => {
+          loadOnServer(renderProps, store).then(() => {
```

Comment:
https://github.com/Rezonans/redux-async-connect/blob/master/modules/ReduxAsyncConnect.js#L44
```js
export function loadOnServer({ components, params }, store, helpers) {
  return Promise.all(asyncConnectPromises(filterAndFlattenComponents(components), params, store, helpers)).then(() => {
    store.dispatch(initialDataLoaded());
  });
}
```

`Promise.all()` return value [resolves when all its arguments are resolved (not rejected)](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise/all); if one of them fails (and that's the user-land code, it can fail), then the resulting promise is rejected. Moreover, it won't do `store.dispatch(initialDataLoaded());` after that.
So, the server will hang indefinitely with incomplete response pending.